sdk/sdk.go

```diff
@@ -0,0 +1,162 @@
+// Package tfsdklog provides helper functions for logging from SDKs and
+// frameworks for building Terraform plugins.
+//
+// Plugin authors shouldn't need to use this package; it is meant for authors
+// of the frameworks and SDKs for plugins. Plugin authors should use the tflog
+// package.
+//
+// This package provides very similar functionality to tflog, except it uses a
+// separate namespace for its logs.
+package tfsdklog
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type loggerKey string
+
+var (
+	rootLoggerKey loggerKey = "sdk"
+	stderr        io.Writer
+)
```

Comment:
I picked `root` over `sdk` because it's in the SDK package, so all loggers are SDK loggers, what makes this one special is it's the root of a tree of sub-loggers and their sub-loggers branched off it.